### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
A bunch of fixes and cleanup that I bumped into while working on the on-demand download PR. See commit messages for details.

### CODE REVIEW

File: pageserver/src/tenant.rs [hunk 1]
```diff
@@ -144,17 +144,20 @@ impl Tenant {
...
             .collect()
     }
 
-    /// Create a new, empty timeline. The caller is responsible for loading data into it
-    /// Initdb lsn is provided for timeline impl to be able to perform checks for some operations against it.
+    ///
+    /// This is used to create the initial 'main' timeline during bootstrapping,
+    /// or when importing a new base backup. The caller is expected to load an
+    /// initial image of the datadir to the new timeline after this.
+    ///
```
Comment (koivunej) [inline pageserver/src/tenant.rs:160]:
```suggestion
    /// This is used to create the initial 'main' timeline during bootstrapping,
    /// or when importing a new base backup. The caller is expected to load an
    /// initial image of the datadir to the new timeline after this.
```

Minor comment nit, unsure what the empty lines before and after are for? Similar with `fn branch_timeline`.

**Review by koivunej — APPROVED** [review]
Minor newlines-in-doc nit only, glad to see at least one familiar comment go :)

File: pageserver/src/tenant/timeline.rs [hunk 2]
```diff
@@ -475,10 +475,6 @@ impl Timeline {
     }
 
     /// Mutate the timeline with a [`TimelineWriter`].
-    ///
-    /// FIXME: This ought to return &'a TimelineWriter, where TimelineWriter
-    /// is a generic type in this trait. But that doesn't currently work in
-    /// Rust: https://rust-lang.github.io/rfcs/1598-generic_associated_types.html
```
[the remark this answers is not in the document]
Comment (hlinnaka) [reply pageserver/src/tenant/timeline.rs:481]: Right, see commit message for this change:
```
    Commit c634cb1d36 removed the trait and changed the function to return
    a &TimelineWriter, as the FIXME said we should do, but forgot to remove
    the FIXME.
```

File: pageserver/src/tenant.rs (hunk 1 above)
Comment (hlinnaka) [reply pageserver/src/tenant.rs]: Removed. 

The extra newlines make the comment and function stand out a little more, but there was no particular reason for that here, and we haven't really used that style elsewhere.

File: pageserver/src/tenant/timeline.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (koivunej) [reply pageserver/src/tenant/timeline.rs:481]: Ack. Had not realized how recently these changes were as I had only heard about them. Resolving.

**Review by SomeoneToIgnore — APPROVED** [review]

**Pull request merged**